frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts

```diff
@@ -33,13 +34,16 @@ export const hideAllExistingProperties = (schema: JSONSchema7) => {
 };
 
 const k8sResourceCapabilityToUISchema = (capability: SpecCapability): UiSchema => {
-  const [, suffix] = capability.match(REGEXP_K8S_RESOURCE_SUFFIX) ?? [];
-  const groupVersionKind = suffix?.replace(/:/g, '~');
+  const [, groupVersionKindToken, selectorToken] =
+    capability.match(REGEXP_K8S_RESOURCE_SUFFIX) ?? [];
+  const groupVersionKind = groupVersionKindToken?.replace(/:/g, '~');
+  const selector = selectorFromString(selectorToken?.replace(/\+/g, ' '));
```

Comment:
On a second glance, we probably don't need to encode/decode space characters using '+'. An x-descriptor with spaces in it is perfectly valid.
```suggestion
  const selector = selectorFromString(selectorToken);
```